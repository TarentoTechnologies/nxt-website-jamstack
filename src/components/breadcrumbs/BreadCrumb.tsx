import { Link } from "gatsby";
import React from "react";

import { breadCrumbLabel } from "../../styles/style-guide/Typography.module.css";
import { breadcrumbStyles } from "./BreadCrumb.module.css";

interface BreadcrumbItem {
  label: any;
  url?: any;
}

interface BreadcrumbsProps {
  items: BreadcrumbItem[];
}

export const BreadCrumb = ({ items }: BreadcrumbsProps) => {
  return (
    <div className={`${breadcrumbStyles} ${breadCrumbLabel}`}>
      {items?.map((item, index) => (
        <React.Fragment key={item.label}>
          {index > 0 && <span>|</span>}
          {item.url ? (
            <Link to={item.url}>{item.label}</Link>
          ) : (
            <span>{item.label}</span>
          )}
        </React.Fragment>
      ))}
    </div>
  );
};
