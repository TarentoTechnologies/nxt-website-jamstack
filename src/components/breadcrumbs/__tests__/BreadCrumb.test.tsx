import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { BreadCrumb } from "../BreadCrumb";

const data = [
  {
    label: "AgencyPortfolio",
    url: "/agency-portfolio",
  },
  {
    label: "Bihar Museum",
    url: "/agency-portfolio/bihar-museum",
  },
];

describe("Test if the BreadCrumb component is working as expected", () => {
  it("<BreadCrumb /> matches snapshot", () => {
    const component = render(<BreadCrumb items={data} />);
    expect(component.container).toMatchSnapshot();
  });

  it("<BreadCrumb /> renders without exploding", () => {
    const component = render(<BreadCrumb items={data} />);
    expect(component.getByTestId("bread-crumb")).toBeInTheDocument();
  });
});
