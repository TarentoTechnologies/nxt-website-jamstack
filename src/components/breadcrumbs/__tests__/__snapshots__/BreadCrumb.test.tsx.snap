// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the BreadCrumb component is working as expected <BreadCrumb /> matches snapshot 1`] = `
<div>
  <div
    class="breadcrumbStyles breadCrumbLabel"
    data-testid="bread-crumb"
  >
    <a
      href="/agency-portfolio"
    >
      AgencyPortfolio
    </a>
    <span>
      |
    </span>
    <a
      href="/agency-portfolio/bihar-museum"
    >
      Bihar Museum
    </a>
  </div>
</div>
`;
